src/solvers/gurobi_solver.py

```diff
@@ -0,0 +1,54 @@
+"""This work was partially funded by the EuroHPC PL project,
+funded in the frame of Smart Growth Operational Programme, topic 4.2.
+
+Authors: Mateusz Hurbol, Justyna Zawalska
+"""
+from typing import Dict
+
+import gurobipy as gp
+from parse import parse
+
+from ..utils.execution_stats import calculate_time
+from .solver import Solver
+
+
+class GurobiSolver(Solver):
+    """Solver that uses Gurobi Optimizer."""
+
+    def solve(self) -> None:
+        """Finds the solution and saves it."""
+        solution = self.find_solution()
+        self.save_result(solution)
+
+    @calculate_time
+    def find_solution(self) -> Dict[str, str]:
+        """Uses Gurobi Optimizer to find the optimal solution."""
+        gpm = gp.Model("workflow")
+        binary_variables = []
+        tasks_amount = len(self.tasks)
```

Comment:
```suggestion
```
I don't think that this variable is helpful. In for loops use `len(self.tasks)`, it's the same amount of characters